core/arch/x86_64/kernel/thread_private.h

```diff
@@ -139,11 +134,31 @@ void thread_sp_alloc_and_run(struct thread_smc_args *args);
  * context of the current thread if THREAD_FLAGS_COPY_ARGS_ON_RETURN is set
  * in the flags field in the thread context.
  */
-void thread_resume(struct thread_ctx_regs *regs);
+void thread_resume(struct thread_ctx86_regs *regs, vaddr_t tmp_stack);
+
+void thread_rpc_resume(vaddr_t sp, struct thread_smc_args *args,
+						vaddr_t tmp_stack);
+/*
+ * Resume thread status from a foreign interrupt
+ */
+void foreign_intr_resume(vaddr_t sp, vaddr_t abt_stack);
+
+/*
+ * Get the return value from a standard secure monitor call
+ */
+void thread_get_stdcall_ret(struct thread_smc_args *args);
+
+/*
+ * Main loop between secure world and non-secure world
+ */
+void sm_sched_nonsecure(void);
 
-uint32_t __thread_enter_user_mode(struct thread_ctx_regs *regs,
-				  uint32_t *exit_status0,
-				  uint32_t *exit_status1);
+uint32_t __thread_enter_user_mode(struct tee_usr_args *args,
+				unsigned long user_sp,
+				unsigned long user_func,
```

Comment:
no, user_sp and user_func are still used. The two items are copied to args before enter to __thread_enter_user_mode assembly code.

Reply:
Can we remove tee_usr_args usage? It's added for x86 specific at that time.  Since Arm code aggregates parameters to thread_ctx_regs including user_sp and user_func, we can also put all of parameters into thread_ctx_regs like Arm code does.

Reply:
yes, we can follow that